docs/source/CUSTOM-JOB.rst

```diff
@@ -0,0 +1,435 @@
+Tutorial: Creating a custom job
+===============================
+
+Rationale
+---------
+
+In the kubernetes environment created by the kiosk, the task of processing images is coordinated by the redis-consumer. The number of consumers at work in any point in time is automatically scaled to match the number of images waiting in a work queue since each redis-consumer can only process one image at a time. Ultimately the redis-consumer is responsible for sending data to tf-serving containers to retrieve model predictions, but it also handles any pre- and post-processing steps that are required by a particular model.
+
+Currently, `deepcell.org <www.deepcell.org>`_ supports a cell tracking feature which is facilitated by the ``tracking-consumer``, which handles the multi-step process of cell tracking:
+
+  1. Send each frame of the dataset for segmentation
+  2. Retrive model predictions and run post-processing to generate cell segmentation masks
+  3. Send cell segmentation masks for cell tracking predictions
+  4. Compile final tracking results and post for download
+
+New data processing pipelines can be implemented by writing a custom consumer. The model can be exported for tf-serving using :func:`~deepcell.utils.export_utils.export_model`.
+
+The following variables will be used throughout the setup of the custom consumer. Pick out names that are appropriate for your consumer.
+
+.. py:data:: queue_name
+
+    Specifies the queue name that will be used to identify jobs for the :mod:`redis_consumer`, e.g. ``'track'``
+
+.. py:data:: consumer_name
+
+    Name of custom consumer, e.g. ``'tracking-consumer'``
+
+.. py:data:: consumer_type
+
+    Name of consumer job, e.g. ``'tracking'``
+
+
+Designing a custom consumer
+---------------------------
+
+Consumers consume redis events. Each type of redis event is put into a separate queue (e.g. ``predict``, ``track``), and each consumer type will pop items to consume off that queue.
+
+Each redis event should have the following fields:
+
+* ``model_name`` - The name of the model that will be retrieved by TensorFlow Serving from ``gs://<bucket-name>/models``
+* ``model_version`` - The version number of the model in TensorFlow Serving
+* ``input_file_name`` - The path to the data file in a cloud bucket.
+
+.. todo::
+
+    Does the user need to set ``model_name`` and ``model_version`` somewhere?
```

Comment:
This is kind of muddled since we used the "automatic label detection".  Currently, `model_name` and `model_version` are only used if `LABEL_DETECT_ENABLED` is false.

Reply:
I think we should document it since presumably people won't be using our label detection models when they design custom consumers. Where do we set those values?

Reply:
Well when they define their custom consumers, I think they will probably not use the `LABEL_DETECT_ENABLED` logic we've created in our consumer.

Ultimately, this is a bug that needs to get resolved on the redis-consumer side, so I think it is not required to document (already documented as issue in redis-consumer itself).

To answer your question, the value is set as an environment variable in the helmfile.